jaeger-core/src/main/java/com/uber/jaeger/Span.java

```diff
@@ -267,19 +267,48 @@ private Span setTagAsObject(String key, Object value) {
   }
 
   @Override
+  public Span log(Map<String, ?> fields) {
+    return log(tracer.clock().currentTimeMicros(), fields);
+  }
+
+  @Override
+  public Span log(long timestampMicroseconds, Map<String, ?> fields) {
+    synchronized (this) {
+      if (context.isSampled()) {
+        if (logs == null) {
+          this.logs = new ArrayList<>();
+        }
+        for (Map.Entry<String, ?> kv : fields.entrySet()) {
+          logs.add(new LogData(timestampMicroseconds, kv.getKey(), kv.getValue()));
```

Comment:
The intended use for jaeger log fields were that each log can be associated with multiple tags. Here, I'm creating a new log for each tag. In the JaegerThriftConverter I can add a special case for when the value is a map and generate multiple tags for the same log.